### ISSUE TITLE
Selecting next song causes "System.NullReferenceException" in NAudio.Core.dll

### ISSUE THREAD

**Issue by Laritello** [Laritello/osu-music#3]
If press "Next" button during the playback you'll get System.NullReferenceException in `SampleAggregator.cs:`
```
public int Read(float[] buffer, int offset, int count)
{
   try
   {
      sampleCount = source.Read(buffer, offset, count);

      for (int n = 0; n < sampleCount; n += channels)
         Add(buffer[n + offset]);

      return sampleCount;
   }
   catch
   {
      return 0;
   }
}
```

Specifically, the line:
`sampleCount = source.Read(buffer, offset, count);`

Maybe it happens because Play method is used right after changing FileStream for the playback. It's not crucial error, but rather annoying.

**Comment by Laritello** [comment]
I've tried to find a solution to this bug, but it seems like I can't fix it without overengineering some stuff. It's not causing any serious problem for the app, so for now I see no point in further investingation.

**Comment by Laritello** [comment]
This issue was fixed with NAudio 2.1.0 update.
